core/arch/arm/kernel/user_ta.c

```diff
@@ -688,6 +689,39 @@ static TEE_Result add_deps(struct user_ta_ctx *utc __unused,
 
 #endif
 
+#ifdef CFG_TA_ASLR
+static size_t aslr_offset(size_t min, size_t max)
+{
+	uint32_t rnd32 = 0;
+	size_t rnd = 0;
+
+	assert(min <= max);
+	if (max > min) {
+		if (crypto_rng_read(&rnd32, sizeof(rnd32)))
+			return 0;
+		rnd = rnd32 % (max - min);
+	}
+	return (min + rnd) * CORE_MMU_USER_CODE_SIZE;
```

Comment:
Actually, this function can be made generic, because it just generates random number between `min` and `max`. How about making it system-wide helper?

Reply:
I'd say it would not make much sense to make it a global helper until it is used in two places at least.

Reply:
That is that we, or someone else remember that this is here already when someone is about to create another function like this. But, this patch set is big enough, so I wouldn't bother at this point.